File: sorts/selection_sort/rb/selection_sort.rb
bigo = [7, 6, 5, 4, 3, 2, 1];
bigomega = [1, 2, 3, 4, 5, 6, 7];

def swap(arr, index_one, index_two)
  temp = arr[index_one]
  arr[index_one] = arr[index_two]
  arr[index_two] = temp
end

def selection_sort(arr)
  for i in 0...arr.length
    min_index = i
    i.upto(arr.length - 1) do |j|
      if arr[j] < arr[min_index]
        min_index = j
      end
    end
    swap(arr, i, min_index) if min_index != i
  end
  return arr
end


puts selection_sort(bigo)
puts selection_sort(bigomega)

File: sorts/selection_sort/js/selection_sort.js
var bigo = [7, 6, 5, 4, 3, 2, 1];
var bigomega = [1, 2, 3, 4, 5, 6, 7];

function swap(arr, indexOne, indexTwo) {
  var temp = arr[indexOne];
  arr[indexOne] = arr[indexTwo];
  arr[indexTwo] = temp;
}

function selectionSort(arr) {
  for(var i = 0; i < arr.length; i++) {
    var minIndex = i;
    for(var j = i; j < arr.length; j++) {
      if(arr[j] < arr[minIndex]) {
        minIndex = j;
      }
    }
    if(minIndex !== i) {
      swap(arr, i, minIndex);
    }
  }
  return arr;
}

console.log(selectionSort(bigo));
console.log(selectionSort(bigomega));
